app/src/main/java/org/mozilla/focus/home/pocket/PocketVideoFragment.kt

```diff
@@ -108,48 +124,68 @@ private class PocketVideoAdapter(
     private val videoItemHorizontalMargin = context.resources.getDimensionPixelSize(R.dimen.pocket_video_item_horizontal_margin)
     private val feedHorizontalMargin = context.resources.getDimensionPixelSize(R.dimen.pocket_feed_horizontal_margin)
 
-    private var feedItems: List<PocketVideo> = if (!feedItemsDeferred.isCompleted) {
-        // We let the parent wait for this to complete to simplify the code here.
-        throw IllegalStateException("Expected deferred to be completed in parent before creation")
-    } else {
-        feedItemsDeferred.getCompleted() ?: emptyList()
+    fun setVideos(videos: List<PocketFeedItem>) {
+        pocketVideos.clear()
+        pocketVideos.addAll(videos)
+        notifyDataSetChanged()
     }
 
-    override fun getItemCount() = feedItems.size
+    override fun getItemCount() = pocketVideos.size
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
             PocketVideoViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.pocket_video_item, parent, false))
 
     override fun onBindViewHolder(holder: PocketVideoViewHolder, position: Int) = with(holder) {
-        val item = feedItems[position]
+        val item = pocketVideos[position]
         setHorizontalMargins(holder, position)
 
-        holder.itemView.setOnClickListener {
-            ScreenController.showBrowserScreenForUrl(holder.itemView.context, fragmentManager, item.url, Session.Source.HOME_SCREEN)
-            TelemetryWrapper.pocketVideoClickEvent(item.id)
+        when (item) {
+            is PocketPlaceholder -> bindPlaceholder(holder)
+            is PocketVideo -> bindPocketVideo(holder, item)
+        }.forceExhaustive
+    }
+
+    private fun bindPlaceholder(holder: PocketVideoViewHolder) {
+        with(holder) {
+            itemView.setOnClickListener(null)
+            itemView.onFocusChangeListener = null
+            titleView.text = ""
+            domainView.text = ""
+            videoThumbnailView.setImageResource(R.color.photonGrey50)
         }
-        holder.itemView.setOnFocusChangeListener { _, hasFocus -> updateForFocusState(holder, hasFocus) }
-        updateForFocusState(holder, holder.itemView.isFocused)
-
-        titleView.text = item.title
-        PicassoWrapper.client.load(item.thumbnailURL).into(videoThumbnailView)
-
-        @Suppress("TooGenericExceptionCaught") // See below.
-        val itemURI = try {
-            URI(item.url)
-        } catch (e: Exception) { // Apparently Kotlin doesn't have multi-catch.
-            when (e) {
-                is URISyntaxException, is NullPointerException -> null
-                else -> throw e
+    }
+
+    private fun bindPocketVideo(holder: PocketVideoViewHolder, item: PocketVideo) {
+        with(holder) {
```

Comment:
nit: Since `bind*` is only likely to use `holder` as a receiver, make this function equal an expression to reduce indentation? It'd better preserve blame too. Here and above.